src/MediaReplacer.php

```diff
@@ -0,0 +1,324 @@
+<?php
+
+namespace P4\MasterTheme;
+
+use WP_Post;
+
+/**
+ * Class MediaReplacer.
+ *
+ * This class is used to replace media elements in WordPress.
+ */
+class MediaReplacer
+{
+    /**
+     * List of image MIME types.
+     * We need this list as for now we will only replace non-image files.
+     */
+    private const IMAGE_MIME_TYPES = [
+        'image/jpeg',
+        'image/png',
+        'image/gif',
+        'image/bmp',
+        'image/webp',
+        'image/tiff',
+        'image/svg+xml',
+    ];
+
+    /**
+     * MediaReplacer constructor.
+     */
+    public function __construct()
+    {
+        add_action('admin_enqueue_scripts', [$this, 'enqueue_media_modal_script']);
+        add_filter('attachment_fields_to_edit', [$this, 'add_replace_media_button'], 10, 2);
+        add_action('add_meta_boxes', [$this, 'add_replace_media_metabox']);
+        add_action('wp_ajax_replace_media', [$this, 'ajax_replace_media']);
+        add_action('admin_notices', [$this, 'display_admin_notices']);
+    }
+
+    /**
+     * Enqueues the JavaScript required for the media replacement modal.
+     */
+    public function enqueue_media_modal_script(): void
+    {
+        wp_enqueue_script(
+            'custom-media-replacer',
+            get_template_directory_uri() . '/admin/js/media_replacer.js',
+            [],
+            Loader::theme_file_ver("admin/js/media_replacer.js"),
+            true
+        );
+    }
+
+    /**
+     * Adds a metabox to the attachments editor.
+     */
+    public function add_replace_media_metabox(): void
+    {
+        add_meta_box(
+            'replace_media_metabox',
+            'Replace Media',
+            [$this, 'render_replace_media_metabox'],
+            'attachment',
+            'side',
+            'low'
+        );
+    }
+
+    /**
+     * Adds a button to replace media files in the attachments editor.
+     *
+     * @param WP_Post $post The current attachment post object.
+     */
+    public function render_replace_media_metabox(WP_Post $post): void
+    {
+        // Check if the post excludes image mime types
+        if (in_array($post->post_mime_type, self::IMAGE_MIME_TYPES)) {
+            echo "<p>Images cannot be replaced yet.</p>";
+            return;
+        }
+
+        echo "<p>Use this to replace the current file without changing the file URL.</p>";
```

Comment:
This can be translatable using the `planet4-master-theme-backend` text domain. Same for all the admin notices further down.

Reply:
Done!